service/controller/app/v1/resource/status/create.go

```diff
@@ -0,0 +1,56 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/controllercontext"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	name := key.AppName(cr)
+
+	ctlCtx, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding status for chart %#q", name))
+
+	chart, err := ctlCtx.G8sClient.ApplicationV1alpha1().Charts(r.watchNamespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return microerror.Maskf(notFoundError, "chart %#q in namespace %#q", name, r.watchNamespace)
```

Comment:
Well, `Get` can return all kinds of errors. Let's please check for the specific error we want to handle and have the catch all as last resort only. What we would do now is to blindly return a very specific error even if the actual error is because of whatever reason.

Reply:
Such as `not found` error you mean?

Reply:
Yes, we should only return `notFoundError` in case it is really a not found error. It does not make sense to return `notFoundError` when there is actually a broken pipe or 500 or whatnot. That only confuses the shit out of people who have to debug something in the mids of a oncall shit storm.